### ISSUE TITLE
App does not realize that contact has Signal

### ISSUE THREAD

**Issue by instagibbs** [WhisperSystems/Signal-Android#4584]
Even though this contact has called me 2 times using encrypted calls, and I've answered, it won't let me initiate any encrypted communication. It treats the contact as a non-Signal user, as far as I can tell.

Ideas?

**Comment by jocelynthode** [comment]
Yeah sometimes it takes a long time to refresh your Signal contacts. If you want to force the refresh earlier you can just click on the icon to write to a new person. From there you should see the list of your Signal contacts. You can then pull-to-refresh. After that everything should work.

**Comment by instagibbs** [comment]
Hmm, tried and the contact didn't appear.

**Comment by jocelynthode** [comment]
Can you write to him using Signal at least ?

**Comment by instagibbs** [comment]
Only un-encrypted. And it prompts me to invite him to the service. But it lists the encrypted call in our history that he made.

**Comment by generalmanager** [comment]
Did you pull down the contact list after clicking on the big round + which means "write new message"? That updates the list of Signal contacts, which usually only happens once a day afair.

**Comment by jocelynthode** [comment]
@generalmanager That's exactly what I suggested and apparently it was not working which is strange because it always fixed these kind of bugs for me.

**Comment by instagibbs** [comment]
Yep, tried that. I was thinking of uninstalling and re-installing to see if that fixes it.

**Comment by jeremymasters** [comment]
Or just unregister and re-register.

**Comment by generalmanager** [comment]
@jocelynthode I read your post, but wanted to clarify that one has to pull down ;-)
@instagibbs as @jeremymasters wrote, unregistering and reregistering should work fine too.

**Comment by instagibbs** [comment]
Re-registering didn't work either.  

Am I just misunderstanding how to add contacts? Everyone else in my contacts were automatically added

**Comment by jeremymasters** [comment]
should all be the same.  Is it possible the other user has stopped using signal and didn't unregister?

**Comment by jocelynthode** [comment]
You shouldn't have anything to do to add contacts, as long as they are in your phone contacts Signal will find them (if you gave the permission to Signal).

Maybe you could try and see if the problem is on your side or on your friend's side. He could try unregistering and register again, etc.

**Comment by i7u** [comment]
@riyapenn the link is truncated.
